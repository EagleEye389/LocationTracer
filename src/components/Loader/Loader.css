.loader-box {
    position: absolute;
    left: 0;
    top: 0;
    background-color: #80808038;
    width: 100%;
    height: 100%;
    z-index: 999;
  }
  .loader-box > .loader {
    position: absolute;
    margin:auto;
    background-color: blue;
    top:0;
    bottom: 0;
    left:0;
    right: 0;
    border-radius: 50%;
    width: 100px;
    height: 100px;
    animation: spin 2s linear infinite;
  }
  
  @keyframes spin {
    0% {
      transform: scale(0.2)
    }
    50%{
      transform: scale(0.5)
    }
    100% {
      transform: scale(0.8);
    }
  }
  