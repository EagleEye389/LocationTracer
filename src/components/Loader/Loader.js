import React from "react";
import PropTypes from 'prop-types';

import "./Loader.css";


/**
 * @name Loader
 * @type {Component}
 * @description Show a loader on screen during api calls or async operations.
 * @param {{isLoading}} Object Props to stateless component
 * @returns {JSX}
 */

const Loader = ({ isLoading }) =>{
  return (
    isLoading && (
      <div className="loader-box">
        <div className="loader" />
      </div>
    )
  );
}

// Prop type need to render this component
Loader.propTypes ={
  isLoading:PropTypes.bool.isRequired
}

export { Loader };
