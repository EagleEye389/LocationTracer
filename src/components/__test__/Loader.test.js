import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Loader from "../loader/Loader";

let component;

describe("Loader test when isLoading props is false ", () => {
  beforeEach(() => {
    component = shallow(<Loader isLoading={false} />);
  });
  it("does not has loader-box", () => {
    expect(component.find(".loader-box").length).toEqual(0);
  });
});

describe("Loader snapshot", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Loader isLoading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("doesnot render", () => {
    const tree = renderer.create(<Loader isLoading={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Loader test when isLoading props is true", () => {
  beforeEach(() => {
    component = shallow(<Loader isLoading={true} />);
  });
  it("has loader-box", () => {
    expect(component.find(".loader-box").length).toEqual(1);
  });
});
