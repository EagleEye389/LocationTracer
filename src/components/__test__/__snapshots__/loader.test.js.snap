// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot doesnot render 1`] = `null`;

exports[`Loader snapshot renders correctly 1`] = `
<div
  className="loader-box"
>
  <div
    className="loader"
  />
</div>
`;
